gcloud-java-datastore/src/test/java/com/google/gcloud/datastore/DatastoreTest.java

```diff
@@ -431,7 +431,7 @@ public void testRunStructuredQuery() {
 
     StructuredQuery<ProjectionEntity> projectionQuery = Query.projectionEntityQueryBuilder()
         .kind(KIND2)
-        .projection(Projection.property("age"), Projection.first("name"))
+        .projection(Projection.property("age"))
         .filter(PropertyFilter.gt("age", 18))
         .groupBy("age")
```

Comment:
The distinct must also be a part of the projection. But yes, you can do a
single column distinct clause.

I think you should drop the group by in favor of distinct on. I think you
are right that you can make the projection only the string name as well. To
make all the properties in the select distinct having a distinct method
with no arguments seems nice. It has a good parallel to the GQL naming.

On Mon, Oct 12, 2015, 7:52 PM Arie <notifications@github.com> wrote:

> In
> gcloud-java-datastore/src/test/java/com/google/gcloud/datastore/DatastoreTest.java
> <https://github.com/GoogleCloudPlatform/gcloud-java/pull/212#discussion_r41822491>
> :
>
> > @@ -431,7 +431,7 @@ public void testRunStructuredQuery() {
> >
> >      StructuredQuery<ProjectionEntity> projectionQuery = Query.projectionEntityQueryBuilder()
> >          .kind(KIND2)
> > -        .projection(Projection.property("age"), Projection.first("name"))
> > +        .projection(Projection.property("age"))
> >          .filter(PropertyFilter.gt("age", 18))
> >          .groupBy("age")
>
> I think we should be able to do a distinct on one column, so having
> Projection.property("name")
> and keeping assertEquals("Dan", entity.getString("name")); should work, no?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-java/pull/212/files#r41822491>
> .
>